hermeznetwork/wallet-ui
# Refactor to typescript slice 06

Closes #684

### What does this PR does?

This slice will start the migration to Typescript of the namespaces `store/transaction` and `views/transaction`.

After analyzing the code on these namespaces it has been concluded that it is important to decouple the different types of transactions supported by the UI. i.e Send, Deposit, and Withdraw, so we end up with a set of [action/reducer/thunks/view] per each type of transaction.

The main reason for this restructuration is that the current code is tangled, complex, and difficult to maintain. 

It's important though that the components are kept generic, so they can be used by the view of any transaction type.

At the store level, this slice will refactor the transactions of type `Send`, creating the following modules in the store (notice the plural in `transactions`):

  * `src/store/transactions/transfer/transfer.actions.ts`
  * `src/store/transactions/transfer/transfer.reducer.ts`
  * `src/store/transactions/transfer/transfer.thunks.ts`

At the presentation level, this slice will only cover the view `transactions/transfer` in charge of managing the transactions of type `Send`. Apart from the view itself, this slice must also include the refactor of the components used by the view.

ℹ️ It's important to note that the old namespaces `store/transaction` and `views/transaction` (singular transaction) **won't be modified at all** in this refactor. This ensures that the types of transactions not covered by each slice will keep working as always.

Once all the new `store` and `views` namespaces are migrated we will be able to completely remove the old namespaces `store/transaction` and `views/transaction`.

### How to test?

In this slice, testing should focus especially on the transactions of type "Send" (a.k.a "Transfer") which are deeply refactored.

## Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [*] Respect code style and lint
